molgenis-data-mysql/src/main/java/org/molgenis/data/mysql/MysqlRepository.java

```diff
@@ -678,7 +678,19 @@ public Entity findOne(Object id)
 	public Iterable<Entity> findAll(Iterable<Object> ids)
 	{
 		if (ids == null) return Collections.emptyList();
-		return findAll(new QueryImpl().in(getEntityMetaData().getIdAttribute().getName(), ids));
+		Query inQuery = new QueryImpl().in(getEntityMetaData().getIdAttribute().getName(), ids);
+		Map<Object, Entity> index = uniqueIndex(newArrayList(findAll(inQuery)), Entity::getIdValue);
```

Comment:
This means the complete set of results gets cached in memory. On the other hand I'd expect that a query for specific ids is limited to manageable batches of rows. Could add batching here to make sure. What do you guys think?